src/orcheo/graph/state.py

```diff
@@ -1,16 +1,17 @@
 """Graph state for the workflow."""
 
 from __future__ import annotations
-from typing import Annotated
+from typing import Annotated, Any
 from langgraph.graph import MessagesState
 
 
 class State(MessagesState):
     """State for the graph."""
 
-    outputs: Annotated[dict, dict_reducer]
+    inputs: Annotated[dict[str, Any], dict_reducer]
```

Comment:
Distinguish these fields better, e.g., by renaming: the current `inputs` are meant for the inputs of the whole workflow, and it doesn't need a reducer; the `outputs` field means the outputs of each workflow node.